### PR TITLE
Writer fixes.

### PR DESCRIPTION
**Pull request by nloppi**

### CODE REVIEW

File: pyfr/integrators/base.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 import re
 import time
 
+import itertools as it
```
Comment (FreddieWitherden) [inline pyfr/integrators/base.py]: Itertools is part of Python.

File: pyfr/plugins/tavg.py [hunk 2]
```diff
@@ -150,5 +151,6 @@ def __call__(self, intg):
 
                 self._writer.write(accmex, metadata, intg.tcurr)
 
-                self.tout = intg.tcurr + self.dtout
```
Comment (FreddieWitherden) [inline pyfr/plugins/tavg.py]: Why are we moving this line?

File: pyfr/plugins/writer.py [hunk 3]
```diff
@@ -19,9 +19,9 @@ def __init__(self, intg, cfgsect, suffix=None):
         self._writer = NativeWriter(intg, self.nvars, basedir, basename,
                                     prefix='soln')
 
-        # Output time step and next output time
+        # Output time step and last output time
         self.dt_out = self.cfg.getfloat(cfgsect, 'dt-out')
-        self.tout_next = intg.tcurr
+        self.tout = intg.tcurr
```
Comment (FreddieWitherden) [inline pyfr/plugins/writer.py]: Rename to tout_last or last_tout? (Same for tavg.)

File: pyfr/plugins/writer.py [hunk 4]
```diff
@@ -32,11 +32,9 @@ def __init__(self, intg, cfgsect, suffix=None):
         # If we're not restarting then write out the initial solution
         if not intg.isrestart:
```
Comment (FreddieWitherden) [inline pyfr/plugins/writer.py]: Double check that behaviour here has not changed (normal run and restart for when start/restart time is/is not a multiple).

Comment (nloppi) [reply pyfr/plugins/writer.py]: The sanity checks seem to work and the tavg fields look right. I tried
1. Fresh start with dt-out that is multiple of dt works
2. Fresh start with dt-out that is not multiple of dt fails
3. Restart with dt-out that is multiple of dt works. (if dt is changed you need to also change dt-out)
4. Restart with dt-out that is not multiple of dt fails

Standard stepper works with any dt-out.

**Pull request merged**